torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -501,6 +501,7 @@ void ProcessGroupAgent::pollTimedOutRPCs() {
 
     if (sleepTime == INFINITE_TIMEOUT) {
       futureTimeoutCV_.wait(lock);
+      continue;
```

Comment:
Not completely sure whether we should have this or not. One case that we should optimize for is when a new future with a shorter timeout is created, this thread should wake up. With the continue, we will just recompute the new sleep time and go back to sleep. Without it, we'll check in `processTimedOutFutures` whether the first future is already timed out - maybe we should keep this behavior?

Reply:
Say it takes time X from `enqueueSend` adding a closer end time request with closer end time to the watchdog thread being waken up and running to get a new `sleepTime`.
The shorter timeout(TTL) is time Y.
I would assume time X is at magnitude of us. While users set timeout at magnitude of second (for practical use cases).
So I believe there is gain here.

Reply:
Never mind. I realized that the below `futureTimeoutCV_.wait_for(lock, sleepTime);` is the critical path.

For most timeout settings, `futureTimeoutCV_.wait(lock);` it used only once at the beginning or a worker has been silent for a long time. 

There is no much gain doing this. 

Reverting the change.